### PR TITLE
(SIMP-5975) Bugfix: Server SSL settings ignored

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Fix bug where some SSL settings could not be set for the puppet server
  `webserver` components.
- Added the following *advanced usage* parameters in case users need to set
  parameters that are not presently managed to work around future issues:
  - pupmod::master::server_webserver_options
  - pupmod::master::ca_webserver_options
  - pupmod::master::extra_webserver_sections
- Temporarily disabled the incron tests until SIMP-5974 can be fixed

SIMP-5975 #close

### CODE REVIEW

File: templates/etc/puppetserver/conf.d/webserver.conf.epp [hunk 1]
```diff
@@ -24,13 +36,35 @@ webserver: {
     client-auth: want
     ssl-crl-path: <%= $pupmod::ssldir %>/crl.pem
```
Comment (jeannegreulich) [inline templates/etc/puppetserver/conf.d/webserver.conf.epp]: Why change the ssldir variable for the machine certs but not the ca and crl?

Comment (trevor-vaughan) [reply templates/etc/puppetserver/conf.d/webserver.conf.epp]: @jeannegreulich They happen to live in different places on the server. It's a bit weird and probably needs to be fully refactored in the future to see if the puppetserver app has pinned down some of the legacy quirks.

Comment (trevor-vaughan) [reply templates/etc/puppetserver/conf.d/webserver.conf.epp]: @jeannegreulich Good question. We probably need to fully refactor the module. This patch is only trying to fix the identified bug with as little potential damage as possible.

**Comment by trevor-vaughan** [comment]
This can be ignored if #95 is closed merged first

File: manifests/master.pp [hunk 2]
```diff
@@ -171,69 +168,144 @@
...
+  Boolean                                             $auditd                          = simplib::lookup('simp_options::auditd', { 'default_value' => false }),
+  Simplib::Port                                       $ca_port                         = simplib::lookup('simp_options::puppet::ca_port', { 'default_value' => 8141 }),
+  Simplib::NetList                                    $trusted_nets                    = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1','::1'] }),
+  String                                              $server_distribution             = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
+  Pupmod::CaTTL                                       $ca_ttl                          = '10y',
+  Boolean                                             $daemonize                       = true,
+  Boolean                                             $enable_ca                       = true,
+  Boolean                                             $enable_master                   = true,
+  Stdlib::AbsolutePath                                $environmentpath                 = $::pupmod::environmentpath,
```
Comment (lnemsick-simp) [inline manifests/master.pp]: What if pupmod class is not included?  I thought we were *not* supposed to set class parameters to values that may not exist?

File: manifests/master.pp [hunk 3]
```diff
@@ -171,69 +168,144 @@
...
+  Boolean                                             $freeze_main                     = false,
+  Simplib::Port                                       $masterport                      = 8140,
+  Stdlib::AbsolutePath                                $puppet_confdir                  = $::pupmod::confdir,
```
Comment (lnemsick-simp) [inline manifests/master.pp]: What if pupmod class is not included?

File: manifests/master.pp [hunk 4]
```diff
@@ -171,69 +168,144 @@
...
+  Stdlib::AbsolutePath                                $confdir                         = $::pupmod::params::master_config['confdir'],
+  Stdlib::AbsolutePath                                $codedir                         = $::pupmod::params::master_config['codedir'],
+  Stdlib::AbsolutePath                                $vardir                          = $::pupmod::params::master_config['vardir'],
+  Stdlib::AbsolutePath                                $rundir                          = $::pupmod::params::master_config['rundir'],
+  Stdlib::AbsolutePath                                $logdir                          = $::pupmod::params::master_config['logdir'],
+  Stdlib::AbsolutePath                                $ssldir                          = $::pupmod::ssldir,
```
Comment (lnemsick-simp) [inline manifests/master.pp]: What if pupmod class is not included?

File: manifests/master.pp (hunk 2 above)
Comment (lnemsick-simp) [reply manifests/master.pp]: This change should be reverted.

File: manifests/master.pp (hunk 3 above)
Comment (lnemsick-simp) [reply manifests/master.pp]: This change should be reverted.

File: manifests/master.pp (hunk 4 above)
Comment (lnemsick-simp) [reply manifests/master.pp]: This change should be reverted.

Comment (lnemsick-simp) [reply manifests/master.pp]: Did not see inheritance change.  This is not a problem.

File: manifests/master.pp (hunk 3 above)
Comment (lnemsick-simp) [reply manifests/master.pp]: Did not see inheritance change.  This is not a problem.

File: manifests/master.pp (hunk 2 above)
Comment (lnemsick-simp) [reply manifests/master.pp]: Did not see inheritance change.  This is not a problem.

**Comment by lnemsick-simp** [comment]
This code was also included in PR #95, which has been merged.

**Pull request closed without merging**